lib/matplotlib/axes/_axes.py

```diff
@@ -3756,8 +3756,7 @@ def apply_mask(arrays, mask):
                     f"'{dep_axis}err' must not contain None. "
                     "Use NaN if you want to skip a value.")
 
-            res = np.zeros(err.shape, dtype=bool)  # Default in case of nan
-            if np.any(np.less(err, -err, out=res, where=(err == err))):
+            if np.any((check := err[err == err]) < -check):
```

Comment:
It's not obvious to me (out of the context of the PR description) what `err == err` is actually doing/checking for here. Could you either do a different, more obvious check where the code conveys what is being checked for, or add a comment explaining what the logic is doing?

Reply:
There was a comment on the line below, but, yes, I have to admit it took me a while to understand what was happening there, so let me expand on that!